src/core/symbology/classification/qgsclassificationmethod.h

```diff
@@ -0,0 +1,290 @@
+    bool symmetricModeAvailable() const {return mSymmetricModeAvailable;}
+
+    /**
+     * Returns if the symmetric mode is enabled
+     */
+    bool symmetricModeEnabled() const {return mSymmetricModeAvailable && mSymmetricEnabled;}
+
+    /**
+     * Returns the symmetry point for symmetric mode
+     */
+    double symmetryPoint() const {return mSymmetryPoint;}
+
+    /**
+     * Returns if the symmetric mode is astride
+     * if TRUE, it will remove the symmetry point break so that the 2 classes form only one
+     */
+    bool symmetryAstride() const {return mSymmetryAstride;}
+
+    /**
+     * Defines if the symmetric mode is enables and configures its parameters.
+     * If the symmetric mode is not available in the current implementation, calling this method has no effect.
+     * \param enabled if the symmetric mode is enabled
+     * \param symmetryPoint the value of the symmetry point
+     * \param symmetryAstride if TRUE, it will remove the symmetry point break so that the 2 classes form only one
+     */
+    void setSymmetricMode( bool enabled, double symmetryPoint = 0, bool symmetryAstride = false );
+
+    // Label properties
+    //! Returns the format of the label for the classes
+    QString labelFormat() const { return mLabelFormat; }
+    //! Defines the format of the labels for the classes, using %1 and %2 for the bounds
+    void setLabelFormat( const QString &format ) { mLabelFormat = format; }
+    //! Returns the precision for the formatting of the labels
+    int labelPrecision() const { return mLabelPrecision; }
+    //! Defines the precision for the formatting of the labels
+    void setLabelPrecision( int labelPrecision );
+    //! Returns if the trailing 0 are trimmed in the label
+    bool labelTrimTrailingZeroes() const { return mLabelTrimTrailingZeroes; }
+    //! Defines if the trailing 0 are trimmed in the label
+    void setLabelTrimTrailingZeroes( bool trimTrailingZeroes ) { mLabelTrimTrailingZeroes = trimTrailingZeroes; }
+
+    //! Transforms a list of classes to a list of breaks
+    static QList<double> listToValues( const QList<QgsClassificationRange> classes );
+
+    /**
+     * This will calculate the breaks for a given layer to define the classes.
+     * The breaks do not contain the uppper and lower bounds (minimum and maximum values).
+     * \param vl The vector layer
+     * \param fieldName The name of the field on which the classes are calculated
+     * \param nclasses The number of classes to be returned
+     */
+    QList<QgsClassificationRange> classes( const QgsVectorLayer *vl, const QString &expression, int nclasses );
+
+    /**
+     * This will calculate the breaks for a list of values.
+     * The breaks do not contain the uppper and lower bounds (minimum and maximum values)
+     * \param values The list of values
+     * \param nclasses The number of classes to be returned
+     */
+    QList<QgsClassificationRange> classes( const QList<double> &values, int nclasses );
```

Comment:
I think there's a missing method here -- there's no way for a caller to get the numeric breaks alone (ie as a list of doubles) from the public api.

Reply:
there's a static method classes to breaks for this.
otherwise, I'd need to duplicate all 3 classes methods